docs/windowing.md

```diff
@@ -0,0 +1,359 @@
+
+# Windowed Aggregations
+
+
+## What Are Windows?
+In stream processing, windows are used to divide endless streams of events into finite time-based intervals.
+
+With windows, you can calculate such aggregations as:
+- Total of website visitors for every hour
+- An average speed of a vehicle over the last 10 minutes
+- Maximum temperature of the sensor for each 30s
+
+## Implementation Details
+
+Here are some general concepts about how windowed aggregations are implemented in Quix Streams:
+
+- Only time-based windows are supported. 
+- Every window is grouped by the current Kafka message key. 
+- The minimal window unit is a **millisecond**.  
+More fine-grained values (e.g. microseconds) will be rounded towards the closest millisecond number.
+
+
+## Types of Time in Streaming
+
+There are two types of time in streaming systems:
```

Comment:
I think the attempt to not mention the time_extractor API makes the part of docs around timestamp very complicated. Maybe we can document it and mentioned that it will be changed in future release and even mention why?

Reply:
Agree, I added the docs about the timestamp extractor.